src/pkg/exporter/js_collector.go

```diff
@@ -0,0 +1,103 @@
+package exporter
+
+import (
+	"github.com/gomodule/redigo/redis"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// JobServiceCollectorName ...
+const JobServiceCollectorName = "JobServiceCollector"
+
+var (
+	jobServiceTaskQueueSize = typedDesc{
+		desc:      newDescWithLables("", "task_queue_size", "Total number of tasks", "type"),
+		valueType: prometheus.GaugeValue,
+	}
+	jobServiceTaskQueueLatency = typedDesc{
+		desc:      newDescWithLables("", "task_queue_latency", "how long ago the next job to be processed was enqueued", "type"),
+		valueType: prometheus.GaugeValue,
+	}
+	jobServiceConcurrency = typedDesc{
+		desc:      newDescWithLables("", "task_concurrency", "Total number of concurrency on a pool", "type", "pool"),
+		valueType: prometheus.GaugeValue,
+	}
+	jobServiceScheduledJobTotal = typedDesc{
+		desc:      newDesc("", "task_scheduled_total", "total number of scheduled job"),
+		valueType: prometheus.GaugeValue,
+	}
+	jobServiceScheduledJobFails = typedDesc{
+		desc:      newDescWithLables("", "task_scheduled_fails", "fail number of a scheduled job", "type"),
+		valueType: prometheus.GaugeValue,
+	}
+)
+
+// NewJobServiceCollector ...
+func NewJobServiceCollector() *JobServiceCollector {
+	return &JobServiceCollector{Namespace: namespace}
+}
+
+// JobServiceCollector ...
+type JobServiceCollector struct {
+	Namespace string
+}
+
+// Describe implements prometheus.Collector
+func (hc *JobServiceCollector) Describe(c chan<- *prometheus.Desc) {
+	c <- jobServiceTaskQueueSize.Desc()
+	c <- jobServiceTaskQueueLatency.Desc()
+	c <- jobServiceConcurrency.Desc()
+	c <- jobServiceScheduledJobFails.Desc()
+}
+
+// Collect implements prometheus.Collector
+func (hc *JobServiceCollector) Collect(c chan<- prometheus.Metric) {
+	for _, m := range hc.getJobserviceInfo() {
+		c <- m
+	}
+}
+
+func (hc *JobServiceCollector) getJobserviceInfo() []prometheus.Metric {
+	if CacheEnabled() {
+		value, ok := CacheGet(JobServiceCollectorName)
+		if ok {
+			return value.([]prometheus.Metric)
+		}
+	}
+	result := []prometheus.Metric{}
+
+	// Get concurrency info via raw redis client
+	rdsConn := GetRedisPool().Get()
+	defer rdsConn.Close()
+	knownJobvalues, err := redis.Values(rdsConn.Do("SMEMBERS", redisKeyKnownJobs(jsNamespace)))
```

Comment:
If possible, I suggest you can put the Redis read/write logic into a separate func for clarification.